nibabel/spatialimages.py

```diff
@@ -866,6 +872,52 @@ def from_image(klass, img):
                      klass.header_class.from_header(img.header),
                      extra=img.extra.copy())
 
+    @classmethod
+    def is_valid_extension(klass, ext):
+        return np.any([ft[1] == ext.lower() for ft in klass.files_types])
+
+    @classmethod
+    def is_image(klass, filename, sniff=None):
```

Comment:
Yes, reasonable names.  Do we want to convey the 'maybe' aspect as we did for the header?